app/post.go

```diff
@@ -2161,14 +2161,19 @@ func (a *App) CheckPostReminders() {
 				continue
 			}
 
+			var permalink string
+			if metadata.TeamName == "" {
+				permalink = fmt.Sprintf("%s/pl/%s", siteURL, postID)
```

Comment:
This was the way it was originally. But inspite of the fact that permalinks work without the team name, we need the team name to appear in the permalink because while the permalink expands the original link, if you click on the link it goes to the "Team Not Found" page.

That's why we decided to implement this in the client side.

Please see the discussion here: https://community.mattermost.com/private-core/pl/8yahcxnb3p8rfb8q6on6jbc9mo

and I believe the decision was to implement this in the mobile app from the last comment: https://community.mattermost.com/private-core/pl/19g87gej4t86pjchty5cw1f5ch

Please feel free to discuss this with @AshishDhama and @harshilsharma63 for more context.

Reply:
We need to build URLs in the mobile app just like we do for permalink.

Reply:
Ok, I see. I missed that context from the conversation. Thank you both!